SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LocalReceptionistProxyServerManager.h"
+
+#include "Internationalization/Regex.h"
+#include "Sockets.h"
+#include "SocketSubsystem.h"
+#include "SpatialCommandUtils.h"
+#include "UObject/CoreNet.h"
+
+DEFINE_LOG_CATEGORY(LogLocalReceptionistProxyServerManager);
+
+#define LOCTEXT_NAMESPACE "FLocalReceptionistProxyServerManager"
+
+FLocalReceptionistProxyServerManager::FLocalReceptionistProxyServerManager()
+: RunningCloudDeploymentName("")
+{
+}
+
+bool FLocalReceptionistProxyServerManager::KillProcessBlockingPort()
+{
+	bool bSuccess = true;
+
+	const FString NetStatCmd = FString::Printf(TEXT("netstat"));
+
+	// -a display active tcp/udp connections, -o include PID for each connection, -n don't resolve hostnames
+	const FString NetStatArgs = TEXT("-n -o -a");
+	FString NetStatResult;
+	int32 ExitCode;
+	FString StdErr;
+	bSuccess = FPlatformProcess::ExecProcess(*NetStatCmd, *NetStatArgs, &ExitCode, &NetStatResult, &StdErr);
+
+	if (ExitCode == ExitCodeSuccess && bSuccess)
+	{
+		// Get the line of the netstat output that contains the port we're looking for.
+		FRegexPattern PidMatcherPattern(FString::Printf(TEXT("(.*?:%i.)(.*)( [0-9]+)"), ReceptionistPort));
+		FRegexMatcher PidMatcher(PidMatcherPattern, NetStatResult);
+		if (PidMatcher.FindNext())
+		{
+			FString Pid = PidMatcher.GetCaptureGroup(3 /* Get the PID, which is the third group. */);
+
+			const FString TaskKillCmd = TEXT("taskkill");
+			const FString TaskKillArgs = FString::Printf(TEXT("/F /PID %s"), *Pid);
+			FString TaskKillResult;
+			bSuccess = FPlatformProcess::ExecProcess(*TaskKillCmd, *TaskKillArgs, &ExitCode, &TaskKillResult, &StdErr);
+			bSuccess = bSuccess && ExitCode == ExitCodeSuccess;
+			if (!bSuccess)
+			{
+				UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to kill process blocking required port. Error: %s"), *StdErr);
+			}
+		}
+		else
+		{
+			bSuccess = false;
+			UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to find PID of the process that is blocking the runtime port."));
+		}
+	}
+	else
+	{
+		bSuccess = false;
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to find the process that is blocking required port. Error: %s"), *StdErr);
+	}
+
+	return bSuccess;
+
+}
+
+bool FLocalReceptionistProxyServerManager::LocalReceptionistProxyServerPreRunChecks()
+{
+	//Check if any process is blocking the 7777 port
+	//if(CheckIfPortIsBound(ReceptionistPort))
```

Comment:
I am not sure if I should add a CheckIfPortIsBound function the one that LocalDeploymentManager does not seem to work as I expected so I am going to add a new one.